web/src/controller/user_controller.rs

```diff
@@ -36,3 +39,34 @@ pub async fn create(
 
     Ok(Json(ApiResponse::new(StatusCode::CREATED.into(), user)))
 }
+
+/// UPDATE a User
+/// NOTE: that this is for updating the current user and as such uses the user
+/// from the AuthenticatedUser extractor. If we decide to allow a user to update
+/// another user, we may want to consider something like a PUT /myself endpoint for
+/// the current user updating their own data.
+#[utoipa::path(
+    put,
+    path = "/users",
+    params(
+        ApiVersion,
+        UpdateUserParams
+    ),
+    request_body = UpdateUserParams,
+    responses(
+        (status = 200, description = "Successfully updated a User", body = [users::Model]),
```

Comment:
What are your thoughts on starting to follow the spec on a successful PUT operation for what HTTP response status code we respond with? It seems like we can respond with a 204, and there's no body returned in a successful update. There's also a 201 response as well for when a `User` could be created in the process of doing a PUT.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

Reply:
Good call out. I went back and forth on this but the fact that you're bringing it up, I think, tips the scales. Update to return 204 in 67be87dc2173cb9da0864924750deaeb8084bd18